src/rgw/rgw_rest_s3.cc

```diff
@@ -6552,10 +6548,12 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
     i++;
   }
   m_aws_response_handler = std::make_unique<aws_response_handler>(s,this);
+
   if (processed_size == s->obj_size) {
+    if (status >=0) {
     m_aws_response_handler.get()->init_stats_response();
     m_aws_response_handler.get()->send_stats_response(processed_size,processed_size,m_aws_response_handler->get_sql_result().size());
```

Comment:
according to spec the bytes-returned should be as followed.
/Total number of bytes of records payload data returned by Amazon S3./